### PR TITLE
Feature/subscription frontend

### PR DESCRIPTION
**Pull request by andmue**

### CODE REVIEW

**Review by mleimer — DISMISSED** [review]

File: frontend/src/components/pages/start/Join.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    forwardToQuiz(joinId: string) {
+        this.props.history.push(`/${joinId}`);
+    }
+
+    render() {
+        return (
+            <div>
+                <Textfield onChange={(e: any) => {this.setState({...this.state, joinId: e.target.value})}}
+                           label="Quiz ID"
+                           pattern="-?[0-9]*(\.[0-9]+)?"
```
Comment (mleimer) [inline frontend/src/components/pages/start/Join.tsx]: Die QuizId erlaubt andere Zeichen [a-z] und [0-9]

File: frontend/src/components/pages/quiz/WaitingRoom.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import React, {Component} from 'react'
+
+export default class WaitingRoom extends Component<any, any> {
```
Comment (danielsuter) [inline frontend/src/components/pages/quiz/WaitingRoom.tsx:3]: why not use shorthand syntax?

File: frontend/src/components/pages/start/Create.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import React, {Component} from 'react'
+
+export default class Create extends Component<any, any> {
```
Comment (danielsuter) [inline frontend/src/components/pages/start/Create.tsx:3]: why not use shorthand syntax?

File: frontend/src/components/pages/start/Join.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+const JOIN_MUTATION = gql`
+    mutation join($joinId: String!){
+        join(joinId: $joinId) {
+          joinId
+        }
+    }
+`;
+
+class Join extends Component<any, any> {
+    state = {
```
Comment (danielsuter) [inline frontend/src/components/pages/start/Join.tsx]: use types

File: frontend/src/components/pages/start/Join.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        joinId: "",
+        nickname: ""
+    };
+
+    constructor(props: any) {
```
Comment (danielsuter) [inline frontend/src/components/pages/start/Join.tsx:21]: use types

File: frontend/src/components/pages/start/Join.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        super(props);
+        this.joinQuiz = this.joinQuiz.bind(this);
+    }
+
+    joinQuiz() {
+        this.props.client.mutate({
+                mutation: JOIN_MUTATION,
+                variables: {
+                    joinId: this.state.joinId,
+                    nickname: this.state.nickname
```
Comment (danielsuter) [inline frontend/src/components/pages/start/Join.tsx:31]: nickname is not defined

File: frontend/src/components/pages/start/Join.tsx (hunk 1 above)
Comment (mleimer) [reply frontend/src/components/pages/start/Join.tsx]: Mit https://github.com/webplatformz/quiz/pull/14 gelten jedoch andere Werte

File: frontend/src/components/pages/start/Join.tsx [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    forwardToQuiz(joinId: string) {
+        this.props.history.push(`/${joinId}`);
+    }
+
+    render() {
+        return (
+            <div>
+                <Textfield onChange={(e: any) => {this.setState({...this.state, joinId: e.target.value})}}
```
Comment (danielsuter) [inline frontend/src/components/pages/start/Join.tsx]: type e

File: frontend/src/components/pages/start/Join.tsx [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+                           label="Quiz ID"
+                           pattern="-?[0-9]*(\.[0-9]+)?"
+                           style={{width: '200px'}} />
+                <br/>
+                <Textfield onChange={(e: any) => {this.setState({...this.state, nickname: e.target.value})}}
```
Comment (danielsuter) [inline frontend/src/components/pages/start/Join.tsx]: type Event

File: frontend/src/components/pages/start/Start.tsx [hunk 9]
```diff
@@ -0,0 +1,14 @@
+import React, {Component} from 'react'
+import Create from "./Create";
+import Join from "./Join";
+
+export default class Start extends Component<any, any> {
```
Comment (danielsuter) [inline frontend/src/components/pages/start/Start.tsx:5]: use shorthand syntax

File: server/domain/player.ts [hunk 10]
```diff
@@ -0,0 +1,5 @@
+export class Player {
+
+    constructor(public id: number, public name: string, public score: number) {
```
Comment (danielsuter) [inline server/domain/player.ts]: use string instead of number

**Review by danielsuter — CHANGES_REQUESTED** [review]

File: frontend/src/components/pages/start/Join.tsx (hunk 1 above)
Comment (mleimer) [reply frontend/src/components/pages/start/Join.tsx]: ALLOWED_CHARS = "ABCDEFGHJKLMNPQRSTUVWXYZabcdefghkmnpqrstuvwxyz123456789";

**Review by mleimer — APPROVED** [review]

**Pull request merged**